# Phase 2 Review

In [1]:
import pandas as pd
import numpy as np
from sklearn.preprocessing import StandardScaler
from statsmodels.formula.api import ols
from scipy import stats
from statsmodels.api import OLS
pd.set_option('display.max_columns', 100)

### Check Your Data … Quickly
The first thing you want to do when you get a new dataset, is to quickly to verify the contents with the .head() method.

In [2]:
df = pd.read_csv('movie_metadata.csv')
print(df.shape)
df.head()

(5043, 28)


color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4    NaN        Doug Walker                     NaN       NaN   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    131.0                     NaN        Rob Walker   

   actor_1_facebook_likes        gross                           genres  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                 40000.0  309404152.0         Action|Adventure|Fantasy   
2                 11000.0  200074175.0        Action|Adventure|Thriller   
3                 27000.0  448130642.0                  Action|Thriller   
4                   131.0          NaN                      Documentary   

      actor_1_name                                        movie_title  \
0      CCH Pounder                                            Avatar    
1      Johnny Depp          Pirates of the Caribbean: At World's End    
2  Christoph Waltz                                           Spectre    
3        Tom Hardy                             The Dark Knight Rises    
4      Doug Walker  Star Wars: Episode VII - The Force Awakens    ...   

   num_voted_users  cast_total_facebook_likes          actor_3_name  \
0           886204                       4834             Wes Studi   
1           471220                      48350        Jack Davenport   
2           275868                      11700      Stephanie Sigman   
3          1144337                     106759  Joseph Gordon-Levitt   
4                8                        143                   NaN   

   facenumber_in_poster                                      plot_keywords  \
0                   0.0             avatar|future|marine|native|paraplegic   
1                   0.0  goddess|marriage ceremony|marriage proposal|pi...   
2                   1.0                bomb|espionage|sequel|spy|terrorist   
3                   0.0  deception|imprisonment|lawlessness|police offi...   
4                   0.0                                                NaN   

                                     movie_imdb_link  num_user_for_reviews  \
0  http://www.imdb.com/title/tt0499549/?ref_=fn_t...                3054.0   
1  http://www.imdb.com/title/tt0449088/?ref_=fn_t...                1238.0   
2  http://www.imdb.com/title/tt2379713/?ref_=fn_t...                 994.0   
3  http://www.imdb.com/title/tt1345836/?ref_=fn_t...                2701.0   
4  http://www.imdb.com/title/tt5289954/?ref_=fn_t...                   NaN   

  language country content_rating       budget  title_year  \
0  English     USA          PG-13  237000000.0      2009.0   
1  English     USA          PG-13  300000000.0      2007.0   
2  English      UK          PG-13  245000000.0      2015.0   
3  English     USA          PG-13  250000000.0      2012.0   
4      NaN     NaN            NaN          NaN         NaN   

   actor_2_facebook_likes  imdb_score  aspect_ratio  movie_facebook_likes  
0                   936.0         7.9          1.78                 33000  
1                  5000.0         7.1          2.35                     0  
2                   393.0         6.8          2.35                 85000  
3                 23000.0         8.5          2.35                164000  
4                    12.0         7.1           NaN                     0

## Question 1

A Hollywood executive wants to know how much an R-rated movie released after 2000 will earn. The data above is a sample of some of the movies with that rating during that timeframe, as well as other movies. How would you go about answering her question? Talk through it theoretically and then do it in code.

What is the 95% confidence interval for a post-2000 R-rated movie's box office gross?

talk through your answer here


In [ ]:
# do it in code here
- Subset the data for movies that are R-rated post 2000. 
- Drop NAs for gross
- Find mean, standard deviation, and size
- Find standard error using sd and size
- Find margin of error
- +/- mean
- Boom 95% confidence interval

In [9]:
x_bar,sigma,n, 

(27648848.437913906, 39088854.942774445, 1208)

In [7]:
# 95% confidence interval
r_post_2000 = df.query("(content_rating == 'R') & (title_year > 2000)").dropna(subset=['gross'])
x_bar = r_post_2000.gross.mean()
sigma = r_post_2000.gross.std()
n = r_post_2000.gross.count()
se = sigma/np.sqrt(n-1)
t_stat = abs(stats.t.ppf((1-.95)/2, df=n-1))
moe = se*t_stat
x_bar-moe, x_bar+moe

(25441437.853607927, 29856259.022219885)

We are 95% confident that the true population mean is 
within this range of (25441437.853607927, 29856259.022219885)

## Question 2a

Your ability to answer the first question has the executive excited and now she has many other questions about the types of movies being made and the differences in those movies budgets and gross amounts.

Read through the questions below and **determine what type of statistical test you should use** for each question and **write down the null and alternative hypothesis for those tests**.

- Is there a relationship between the number of Facebook likes for a cast and the box office gross of the movie?
- Do foreign films perform differently at the box office than non-foreign films?
- Of all movies created are 40% rated R?
- Is there a relationship between the language of a film and the content rating (G, PG, PG-13, R) of that film?
- Is there a relationship between the content rating of a film and its budget? 

In [ ]:
# your answers here
- Linear Regression
    -H0: Box office gross stays the same depending on likes for cast
    -H1: Box office gross doesn't stay the same depending on likes for cast
- Two sample independent T test
    -H0: mean(foreign) == mean(domestic)
    -H1: mean(foreign) != mean(foreign)
- Z Test for proportions
    -H0: Proportion == .40
    -H1: Proportion != .40
- Chi Squared
    -H0: Distributions are equal
    -H1: Distributions aren't equal
- ANOVA
    -H0: Budget isn't affected by content rating
    -H1: Budget is affected by content rating


## Question 2b

Calculate the answer for the second question:

- Do foreign films perform differently at the box office than non-foreign films?

In [11]:
# your answer here
stats.ttest_ind(df.query("country == 'USA'")['gross'], df.query("country != 'USA'")['gross'], nan_policy='omit')


Ttest_indResult(statistic=12.098302287742106, pvalue=3.8631094668570096e-33)

THERE IS A STATISTICAL SIGNIFICANT DIFFERENCE BETWEEN THE BOX OFFICE GROSS OF FOREIGN AND DOMESTIC FILMS

## Question 3

Now that you have answered all of those questions, the executive wants you to create a model that predicts the money a movie will make if it is released next year in the US. She wants to use this to evaluate different scripts and then decide which one has the largest revenue potential. 

Below is a list of potential features you could use in the model. Would you use all of these features in the model? Identify which features you might drop and why.


*Remember you want to be able to use this model to predict the box office gross of a film **before** anyone has seen it.*

- **budget**: The amount of money spent to make the movie
- **title_year**: The year the movie first came out in the box office
- **years_old**: How long has it been since the movie was released
- **genre**: Each movie is assigned one genre category like action, horror, comedy
- **avg_user_rating**: This rating is taken from Rotten tomatoes, and is the average rating given to the movie by the audience
- **actor_1_facebook_likes**: The number of likes that the most popular actor in the movie has
- **total_cast_facebook_likes**: The sum of likes for the three most popular actors in the movie
- **language**: the original spoken language of the film


In [12]:
df.columns

Index(['color', 'director_name', 'num_critic_for_reviews', 'duration',
       'director_facebook_likes', 'actor_3_facebook_likes', 'actor_2_name',
       'actor_1_facebook_likes', 'gross', 'genres', 'actor_1_name',
       'movie_title', 'num_voted_users', 'cast_total_facebook_likes',
       'actor_3_name', 'facenumber_in_poster', 'plot_keywords',
       'movie_imdb_link', 'num_user_for_reviews', 'language', 'country',
       'content_rating', 'budget', 'title_year', 'actor_2_facebook_likes',
       'imdb_score', 'aspect_ratio', 'movie_facebook_likes'],
      dtype='object')

In [16]:
df['genres'].value_counts()

Drama                                                    236
Comedy                                                   209
Comedy|Drama                                             191
Comedy|Drama|Romance                                     187
Comedy|Romance                                           158
                                                        ... 
Animation|Comedy|Fantasy|Musical                           1
Action|Adventure|Comedy|Family|Fantasy|Mystery|Sci-Fi      1
Action|Adventure|Animation|Family                          1
Crime|Documentary|Drama                                    1
Adventure|Documentary|Drama|Sport                          1
Name: genres, Length: 914, dtype: int64

In [14]:
df['years_old'] = 2021 - df['title_year']
df3 = df[['budget','title_year','years_old','genres','imdb_score', 'actor_1_facebook_likes','cast_total_facebook_likes','language']]

In [18]:
abs(df3.corr())>.75

budget  title_year  years_old  imdb_score  \
budget                       True       False      False       False   
title_year                  False        True       True       False   
years_old                   False        True       True       False   
imdb_score                  False       False      False        True   
actor_1_facebook_likes      False       False      False       False   
cast_total_facebook_likes   False       False      False       False   

                           actor_1_facebook_likes  cast_total_facebook_likes  
budget                                      False                      False  
title_year                                  False                      False  
years_old                                   False                      False  
imdb_score                                  False                      False  
actor_1_facebook_likes                       True                       True  
cast_total_facebook_likes                    True                       True

In [ ]:
"""
Remove avg_user_rating since that happens after a movie is released
Genre's too difficult to deal for get_dummies
Remove either cast_total_facebook_likes or actor_1_facebook_likes -> highly correlated
Remove either years_old or title_year -> function of one another
"""

## Question 4a

Create the following variables:

- `years_old`: The number of years since the film was released.
- Dummy categories for each of the following ratings:
    - `G`
    - `PG`
    - `R`
    
Once you have those variables, create a summary output for the following OLS model:

`gross~cast_total_facebook_likes+budget+years_old+G+PG+R`

In [24]:
pd.get_dummies(df4['content_rating'])

Approved  G  GP  M  NC-17  Not Rated  PG  PG-13  Passed  R  TV-14  TV-G  \
0            0  0   0  0      0          0   0      1       0  0      0     0   
1            0  0   0  0      0          0   0      1       0  0      0     0   
2            0  0   0  0      0          0   0      1       0  0      0     0   
3            0  0   0  0      0          0   0      1       0  0      0     0   
4            0  0   0  0      0          0   0      0       0  0      0     0   
...        ... ..  .. ..    ...        ...  ..    ...     ... ..    ...   ...   
5038         0  0   0  0      0          0   0      0       0  0      0     0   
5039         0  0   0  0      0          0   0      0       0  0      1     0   
5040         0  0   0  0      0          0   0      0       0  0      0     0   
5041         0  0   0  0      0          0   0      1       0  0      0     0   
5042         0  0   0  0      0          0   1      0       0  0      0     0   

      TV-MA  TV-PG  TV-Y  TV-Y7  Unrated  X  
0         0      0     0      0        0  0  
1         0      0     0      0        0  0  
2         0      0     0      0        0  0  
3         0      0     0      0        0  0  
4         0      0     0      0        0  0  
...     ...    ...   ...    ...      ... ..  
5038      0      0     0      0        0  0  
5039      0      0     0      0        0  0  
5040      0      0     0      0        0  0  
5041      0      0     0      0        0  0  
5042      0      0     0      0        0  0  

[5043 rows x 18 columns]

In [34]:
# your answer here
df4 = df[['gross','budget','cast_total_facebook_likes','years_old','content_rating']]
df4wDumms = pd.get_dummies(df4, columns = ['content_rating']).drop('content_rating_PG-13', axis=1)
ols(formula = "gross~cast_total_facebook_likes+budget+years_old+content_rating_G+content_rating_PG+content_rating_R", data=df4wDumms).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  gross   R-squared:                       0.134
Model:                            OLS   Adj. R-squared:                  0.132
Method:                 Least Squares   F-statistic:                     99.95
Date:                Wed, 26 May 2021   Prob (F-statistic):          2.71e-117
Time:                        23:05:23   Log-Likelihood:                -75517.
No. Observations:                3891   AIC:                         1.510e+05
Df Residuals:                    3884   BIC:                         1.511e+05
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
=============================================================================================
                                coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------
Intercept                   5.24e+07   2.62e+06     20.018      0.000    4.73e+07    5.75e+07
cast_total_facebook_likes   864.9426     55.572     15.564      0.000     755.990     973.895
budget                        0.0270      0.005      5.743      0.000       0.018       0.036
years_old                 -1.444e+05   1.06e+05     -1.365      0.172   -3.52e+05     6.3e+04
content_rating_G           2.888e+07   7.06e+06      4.089      0.000     1.5e+07    4.27e+07
content_rating_PG          1.573e+07    3.2e+06      4.913      0.000    9.45e+06     2.2e+07
content_rating_R          -2.851e+07    2.3e+06    -12.378      0.000    -3.3e+07    -2.4e+07
==============================================================================
Omnibus:                     2340.243   Durbin-Watson:                   1.061
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            39932.290
Skew:                           2.544   Prob(JB):                         0.00
Kurtosis:                      17.846   Cond. No.                     1.55e+09
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.55e+09. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## Question 4b

Below is the summary output you should have gotten above. Identify any key takeaways from it.
- How ‘good’ is this model?
- Which features help to explain the variance in the target variable? 
    - Which do not? 


In [ ]:
It isn't very good since it has an extremely low R-squared score meaning that its barely beating the baseline

G doesn't help explain the variance in the target variable. Everything else does. 

<img src="ols_summary.png" style="withd:300px;">

your answer here


## Question 5

**Bayes Theorem**

An advertising executive is studying television viewing habits of married men and women during prime time hours. Based on the past viewing records he has determined that during prime time wives are watching television 60% of the time. It has also been determined that when the wife is watching television, 40% of the time the husband is also watching. When the wife is not watching the television, 30% of the time the husband is watching the television. Find the probability that if the husband is watching the television, the wife is also watching the television.

In [1]:
# your answer here
p_w = .6
p_h_given_w = .4
p_h_given_not_w = .3
p_h = (p_h_given_w*p_w)+(p_h_given_not_w*(1-p_w))

(p_h_given_w*p_w)/p_h

0.6666666666666666

In [35]:
(.4*.6)+(.3*.4)

0.36

In [36]:
(.4*.6)/.36

0.6666666666666666

## Question 6

Explain what a Type I error is and how it relates to the significance level when doing a statistical test. 

In [ ]:
Type 1 error is when you reject the null hypothesis when you shouldn't have. 

Type one error directly relates to significance levels because a an increase in signifcance
levels directly increases the likelihood of type one errors and vice versa. 

If we have a 95% significance level, then we have a 5% chance of making type 1 errors.

## Question 7

How is the confidence interval for a sample related to a one sample t-test?